### PR TITLE
review

### PR DESCRIPTION
**Pull request by bbguimaraes**
@jupierce, @stevekuznetsov.

### CODE REVIEW

File: merge-and-build.sh [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 set -o errexit   # All non-zero statuses will terminate script
 set -o pipefail  # All components of piped command will terminate script if they fail
+set -o nounset   # Makes substituting unset variables an error
```
Comment (bbguimaraes) [inline merge-and-build.sh:5]: These three usually go together.

File: merge-and-build.sh [hunk 2]
```diff
@@ -45,22 +46,21 @@ done
 
 if [[ "${help}" == "1" ]]; then
     usage
-    exit 1
+    exit 0
```
Comment (bbguimaraes) [inline merge-and-build.sh:49]: Exit code for `-h/--help` is normally `0`.

File: merge-and-build.sh [hunk 3]
```diff
@@ -76,11 +76,11 @@ git merge master -m "Merge master into enterprise-${ose_version}"
 cd "${workpath}"
 git clone git@github.com:openshift/ose.git
 cd ose
-git add remote upstream git@github.com:openshift/origin.git
-git fetch --all
+git remote add upstream git@github.com:openshift/origin.git
+git fetch upstream master
```
Comment (bbguimaraes) [inline merge-and-build.sh:80]: This should speed things up by avoiding fetching anything from ose (which we just cloned) and other refs from origin.

File: merge-and-build.sh [hunk 4]
```diff
@@ -76,11 +76,11 @@ git merge master -m "Merge master into enterprise-${ose_version}"
...
 git merge -m "Merge remote-tracking branch upstream/master" upstream/master
 # Pull in the origin-web-console stuff
-vc_commit="$(GIT_REF=master hack/vendor-console.sh 2>/dev/null | grep "Vendoring origin-web-console" | awk '{print $4}')"
+vc_commit=$(GIT_REF=master hack/vendor-console.sh | awk '/Vendoring origin-web-console/{print $4}')
```
Comment (bbguimaraes) [inline merge-and-build.sh:83]: `2>/dev/null` might make debugging errors in `hack/vendor-console.sh` difficult (and it doesn't influence the output capture for `vc_commit`).  Also simplifying grep/awk to just awk.

File: merge-and-build.sh [hunk 5]
```diff
@@ -45,22 +46,21 @@ done
...
     usage
-    exit 1
+    exit 0
 fi
 
 if [[ -z "${major}" || -z "${minor}" ]]; then
-    echo "--major and --minor are required"
-    usage
+    echo >&2 "--major and --minor are required"
+    usage >&2
```
Comment (bbguimaraes) [inline merge-and-build.sh:54]: Sending error output to `stderr`.

File: merge-and-build.sh [hunk 6]
```diff
@@ -45,22 +46,21 @@ done
...
 set -o xtrace  # Verbose script execution output
 ose_version="${major}.${minor}"
 
-buildpath="${HOME}/go"
-cd "$buildpath"
-GOPATH="$( pwd )"; export GOPATH
-workpath="${buildpath}/src/github.com/openshift/"
+GOPATH=${HOME}/go
+export GOPATH
+workpath=${GOPATH}/src/github.com/openshift/
```
Comment (bbguimaraes) [inline merge-and-build.sh:63]: Should be enough.

File: merge-and-build.sh [hunk 7]
```diff
@@ -76,11 +76,11 @@ git merge master -m "Merge master into enterprise-${ose_version}"
 cd "${workpath}"
 git clone git@github.com:openshift/ose.git
 cd ose
-git add remote upstream git@github.com:openshift/origin.git
-git fetch --all
+git remote add upstream git@github.com:openshift/origin.git
```
Comment (bbguimaraes) [inline merge-and-build.sh:79]: Typo.

**Comment by bbguimaraes** [comment]
Can't add a comment to [line 92](https://github.com/jupierce/aos-cd-build-jobs/pull/2/files#diff-b2c89e6e30e500f6b3e2644f11a8b35cR92) because it's not in the diff:

Do we need to export VERSION?  It would be good to document why/where it is needed.

**Comment by jupierce** [comment]
@bbguimaraes the VERSION export wasn't clear to me either. Perhaps something to investigate when you replicate the buildvm environment. In this meantime, your PR looks good, so I will merge.

File: merge-and-build.sh (hunk 4 above)
Comment (stevekuznetsov) [reply merge-and-build.sh:83]: Why getting rid of quotes? Quotes are always good :)

**Comment by bbguimaraes** [comment]
Non-intentional, I probably dropped them while editing.

This is one of the few cases where they don't matter, but I agree it's
best to just put them everywhere.

edit: thanks, github.  This was supposed to be a reply to @stevekuznetsov's comment.

**Comment by jupierce** [comment]
@bbguimaraes @stevekuznetsov Thanks for the reviews. Moving this file into https://github.com/jupierce/aos-cd-jobs/tree/build/ose . Any additional PRs can go there.

**Pull request merged**